src/Credentials/AppIdentityCredentials.php

```diff
@@ -104,11 +112,27 @@ public function fetchAuthToken(callable $httpHandler = null)
     $scope = is_array($this->scope) ? $this->scope : explode(' ', $this->scope);
 
     $token = AppIdentityService::getAccessToken($scope);
+    $this->lastReceivedToken = $token;
 
     return $token;
   }
 
   /**
+   * Implements FetchAuthTokenInterface#getLastReceivedToken.
+   */
+  public function getLastReceivedToken()
+  {
+    if ($this->lastReceivedToken) {
+      return [
+        'access_token' => $this->lastReceivedToken['access_token'],
```

Comment:
How does this work with our caching paradigm? Since we're essentially caching this now. I guess it's alright?